capture/plugins/writer-s3.c

```diff
@@ -607,15 +623,49 @@ void writer_s3_init(char *UNUSED(name))
     s3StorageClass        = moloch_config_str(NULL, "s3StorageClass", "STANDARD");
     s3MaxConns            = moloch_config_int(NULL, "s3MaxConns", 20, 5, 1000);
     s3MaxRequests         = moloch_config_int(NULL, "s3MaxRequests", 500, 10, 5000);
+    s3Token               = NULL;
 
     if (!s3Bucket) {
         printf("Must set s3Bucket to save to s3\n");
         exit(1);
     }
 
-    if (!s3AccessKeyId) {
-        printf("Must set s3AccessKeyId to save to s3\n");
-        exit(1);
+    if (!s3AccessKeyId || !s3AccessKeyId[0]) {
+        // Fetch the data from the EC2 metadata service
+        size_t rlen;
+        void *metadataServer = moloch_http_create_server("http://169.254.169.254", 10, 10, 0);
```

Comment:
What is this ip?  Should it be configurable?

Reply:
This is the AWS metadata URL -- all EC2 machines can access this URL to get metadata about the environment.